src/Storage/Storage/Blob/StorageCloudBlobCmdletBase.cs

```diff
@@ -656,7 +656,7 @@ internal DataLakeFileSystemClient GetFileSystemClientByName(IStorageBlobManageme
             }
             else if (localChannel.StorageContext.StorageAccount.Credentials.IsSAS) //SAS
             {
-                fileSystem = new DataLakeFileSystemClient(new Uri (fileSystemUri.ToString() + localChannel.StorageContext.StorageAccount.Credentials.SASToken));
+                fileSystem = new DataLakeFileSystemClient(new Uri (fileSystemUri.ToString() + "?" + Util.GetSASStringWithoutQuestionMark(localChannel.StorageContext.StorageAccount.Credentials.SASToken)));
```

Comment:
This is just a wild question, is it possible `fileSystemUri` already contains a `?`

Reply:
File system is a container, and container won't have snapshot or version, so the Uri won't have "?"